Makefile

```diff
@@ -256,8 +256,18 @@ include lib/stack_protector/stack_protector.mk
 # makefile may use all previous definitions in this file)
 ################################################################################
 
+# Exclude coreboot support by default, override by command line or plat file
```

Comment:
Yeah, the way I want to use it, it does.

The thing is that coreboot isn't really a per-platform constant. Some platforms support booting from coreboot, but they may still be booted from a different BL2 implementation as well. For example, Rockchip's RK3399 supports booting from coreboot or U-Boot, I think.

I want to add this constant here and then add COREBOOT=1 to the make invocation command line that the coreboot build system uses to build Trusted Firmware. This way, if you build for coreboot it will always automatically be 1, and if you build standalone it will always automatically be 0. You don't have to set anything manually anymore.